ssl/ssl_cert.c

```diff
@@ -73,11 +87,18 @@ CERT *ssl_cert_new(void)
 CERT *ssl_cert_dup(CERT *cert)
 {
     CERT *ret = OPENSSL_zalloc(sizeof(*ret));
-    int i;
+    size_t i;
 
     if (ret == NULL)
         return NULL;
 
+    ret->ssl_pkey_num = cert->ssl_pkey_num;
+    ret->pkeys = OPENSSL_zalloc(ret->ssl_pkey_num * sizeof(CERT_PKEY));
+    if (ret->pkeys == NULL) {
+        ERR_raise(ERR_LIB_SSL, ERR_R_MALLOC_FAILURE);
```

Comment:
With #19301 in the code, there is no need to raise malloc failures on error paths.

Reply:
so changed in several locations in https://github.com/openssl/openssl/pull/19312/commits/e5582173f0db49a28edf0907db053ea66e10d3dc